test/distributed/org/apache/cassandra/distributed/impl/Instance.java

```diff
@@ -443,25 +457,31 @@ public void startup(ICluster cluster)
                 StorageService.instance.registerDaemon(CassandraDaemon.getInstanceForTesting());
                 if (config.has(GOSSIP))
                 {
+                    MigrationManager.setUptimeFn(() -> System.currentTimeMillis() - startedAt);
                     StorageService.instance.initServer();
                     StorageService.instance.removeShutdownHook();
                     Gossiper.waitToSettle();
                 }
                 else
                 {
-                    initializeRing(cluster);
-                }
+                    cluster.stream().forEach(peer -> {
+                        // TODO: this isn't going to work for UpgradableCluster
+                        GossipHelper.statusToNormal((IInvokableInstance) peer).accept(this);
+                    });
 
-                StorageService.instance.ensureTraceKeyspace();
+                }
 
                 SystemKeyspace.finishStartup();
 
                 CassandraDaemon.getInstanceForTesting().setupCompleted();
 
                 if (config.has(NATIVE_PROTOCOL))
                 {
+                    // Start up virtual table support
+                    CassandraDaemon.getInstanceForTesting().setupVirtualKeyspaces();
```

Comment:
we do this on line 424 already, was there a reason to need it here?

Reply:
This wasn't there at the time of writing; removed in favour of the other one